test/hotspot/jtreg/runtime/cds/DeterministicDump.java

```diff
@@ -50,8 +50,13 @@ public static void main(String[] args) throws Exception {
     public static void doTest(boolean compressed) throws Exception {
         ArrayList<String> baseArgs = new ArrayList<>();
 
-        // Use the same heap size as make/Images.gmk
+        // Try to reduce indeterminism of GC heap sizing and evacuation.
         baseArgs.add("-Xmx128M");
+        baseArgs.add("-Xms128M");
+        baseArgs.add("-Xmn128M");
```

Comment:
Is it legal/useful to set `Xmn` = `Xmx`?

Reply:
The GC will just ignore you. It is useful to set it to a high value since it avoids any GC until the CDS dumps happen, removing randomness.

I changed the code to have a slightly smaller `-Xmn` than `-Xms`, and removed all other options. This is sufficient to avoid any GCs until CDS dump time; I do not want to dive into how to make G1 GCs as deterministic as possible here if there are no GCs at all.

(The options are renmants from earlier attempts to make the CDS dump as deterministic as possible in the presence of GCs. After not succeeding (but making issues rare), checking with @iklam showed that it is not the point to make GC deterministic, but it is fine to just pass the test).